homeassistant/components/ptdevices/coordinator.py

```diff
@@ -0,0 +1,119 @@
+"""Cordinator for PTDevices integration."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any, Final
+
+import aioptdevices
+from aioptdevices.interface import PTDevicesResponse
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.debounce import Debouncer
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import (
+    REQUEST_REFRESH_DEFAULT_IMMEDIATE,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+from .const import DOMAIN, UPDATE_INTERVAL
+from .device import ptdevices_get_data
+
+_LOGGER = logging.getLogger(__name__)
+REFRESH_COOLDOWN: Final = 10
+
+type PTDevicesConfigEntry = ConfigEntry[PTDevicesCoordinator]
```

Comment:
I didn't see this with my earlier comment, ensure you are using this type rather than the base ConfigEntry type.